### PR TITLE
Support grpc headers configuration of opentelemetry provider

### PR DESCRIPTION
**Pull request by Kuromesi**
**Please provide a description of this PR:**

This PR aims to support this feature request https://github.com/istio/istio/issues/52873.

I have already tested this implementation on Alibaba Cloud OpenTelemetry, which requires adding header like `Authentication: token-xxxxx` on gRPC requests for authentication and it works fine on my cluster.

### CODE REVIEW

File: pilot/pkg/networking/core/tracing.go [hunk 1]
```diff
@@ -269,17 +269,19 @@ func otelConfig(serviceName string, otelProvider *meshconfig.MeshConfig_Extensio
 		ServiceName: serviceName,
 	}
 
-	if otelProvider.GetHttp() == nil {
-		// export via gRPC
+	if otelProvider.GetGrpc() != nil {
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pilot/pkg/networking/core/tracing.go:272]: (though you would want to make buildInitialMetadata return nil if the input is nil -- probably something we want anyways, though)

File: pilot/pkg/networking/core/tracing.go [hunk 2]
```diff
@@ -269,17 +269,19 @@ func otelConfig(serviceName string, otelProvider *meshconfig.MeshConfig_Extensio
 		ServiceName: serviceName,
 	}
```
Comment (howardjohn) [inline pilot/pkg/networking/core/tracing.go:271]: Can we add to ValidateMeshConfig a check that grpc and http are not both set?

File: pilot/pkg/networking/core/tracing.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kuromesi) [reply pilot/pkg/networking/core/tracing.go:272]: Sure! I just want to ensure that when both `grpc` and `http` are set, traces would still be exported via grpc. ValidateMeshConfig could solve this concern, I will add that!

File: pilot/pkg/networking/core/tracing.go [hunk 3]
```diff
@@ -717,3 +720,15 @@ func buildHTTPHeaders(headers []*meshconfig.MeshConfig_ExtensionProvider_HttpHea
...
 	return target
 }
+
+func buildInitialMetadata(metadata []*meshconfig.MeshConfig_ExtensionProvider_HttpHeader) []*core.HeaderValue {
+	target := make([]*core.HeaderValue, 0, len(metadata))
```
Comment (howardjohn) [inline pilot/pkg/networking/core/tracing.go:725]: This is always setting `initialmetadata` to non nil, can we add `if metadata == nil { return nil }`?

**Review by howardjohn — APPROVED** [review]

File: pilot/pkg/networking/core/tracing.go (hunk 3 above)
Comment (Kuromesi) [reply pilot/pkg/networking/core/tracing.go:728]: Done. :)

**Review by zirain — APPROVED** [review]

**Comment by erikwoo** [comment]
Does GPRC supports configurable path? 
Considering upload tracing data to quickwit，which exposes a GRPC api with path = /api/v1/otlp/v1/traces

**Comment by zirain** [comment]
Envoy gRPC doesn't suppot that, but support it with HTTP protocol.

https://istio.io/latest/docs/reference/config/istio.mesh.v1alpha1/#MeshConfig-ExtensionProvider-OpenTelemetryTracingProvider

**Pull request merged**